.gitignore

```diff
@@ -1,7 +1,43 @@
+# IDE/Filesystem
+.idea
+.vscode
+.DS_Store
+
 *.swp
 *.swo
 *.gcno
 *.gcda
 *.kdev4
 /.kdev4
+
+# Python
 __pycache__
+*.egg-info
+.venv
+build/
+
+# Build outputs
+bytes/*.cpp
+*.so
+*.o
+*.a
+
+# Test artifacts
+tests/*.tok.*
+tests/*.src.*
+tests/*.err
+tests/tests
+
+# CMake/Ninja artifacts
+*.cmake
+cmake-build-debug/
+CMakeFiles/
+Testing/
+CMakeCache.txt
+build.ninja
+.ninja_deps
+.ninja_log
+
+# Executables
+pycdc
+pycdas
```

Comment:
Rather than generating build output in the source tree, why not configure CMake to use a build directory inside Python's build temp dir?

Reply:
Ah, you're right, it's actually just related to my setup since I used the root dir as build output. I'll remove this.